tools/bootstrap_commitlogs/main/main.go

```diff
@@ -0,0 +1,233 @@
+	commitlogsrc "github.com/m3db/m3db/storage/bootstrap/bootstrapper/commitlog"
+	"github.com/m3db/m3db/storage/bootstrap/result"
+	"github.com/m3db/m3db/storage/namespace"
+	"github.com/m3db/m3db/ts"
+	"github.com/m3db/m3db/x/io"
+	"github.com/m3db/m3x/instrument"
+	xlog "github.com/m3db/m3x/log"
+	xtime "github.com/m3db/m3x/time"
+)
+
+var (
+	pathPrefixArg           = flag.String("path-prefix", "/var/lib/m3db", "Path prefix - must contain a folder called 'commitlogs'")
+	namespaceArg            = flag.String("namespace", "metrics", "Namespace")
+	blockSizeArg            = flag.Duration("block-size", 10*time.Minute, "Block size")
+	flushSizeArg            = flag.Int("flush-size", 524288, "Flush size of commit log")
+	bootstrapRetentionArg   = flag.Duration("retention", 48*time.Hour, "Retention")
+	shardsCountArg          = flag.Int("shards-count", 8192, "Shards count - set number too bootstrap all shards in range")
+	shardsArg               = flag.String("shards", "", "Shards - set comma separated list of shards")
+	debugListenAddressArg   = flag.String("debug-listen-address", "", "Debug listen address - if set will expose pprof, i.e. ':8080'")
+	currentUnixTimestampArg = flag.String("current-unix-timestamp", currentUnixTimestampString(), "Current unix timestamp - If set will perform the bootstrap as if this was the current time, defaults to current time")
+)
+
+func currentUnixTimestampString() string {
+	unixTs := time.Now().Unix()
+	return strconv.Itoa(int(unixTs))
+}
+
+func main() {
+	flag.Parse()
+	if *pathPrefixArg == "" ||
+		*namespaceArg == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	var (
+		pathPrefix           = *pathPrefixArg
+		namespaceStr         = *namespaceArg
+		blockSize            = *blockSizeArg
+		flushSize            = *flushSizeArg
+		bootstrapRetention   = *bootstrapRetentionArg
+		shardsCount          = *shardsCountArg
+		shards               = *shardsArg
+		debugListenAddress   = *debugListenAddressArg
+		currentUnixTimestamp = *currentUnixTimestampArg
+	)
+
+	log := xlog.NewLogger(os.Stderr)
+
+	if debugListenAddress != "" {
+		go func() {
+			log.Infof("starting debug listen server at '%s'\n", debugListenAddress)
+			err := http.ListenAndServe(debugListenAddress, http.DefaultServeMux)
+			if err != nil {
+				log.Fatalf("could not start debug listen server at '%s': %v", debugListenAddress, err)
+			}
+		}()
+	}
+
+	shardTimeRanges := result.ShardTimeRanges{}
+
+	currentUnixTimestampParsed, err := strconv.Atoi(currentUnixTimestamp)
+	if err != nil {
+		log.Fatalf("could not parse unix timestamp: '%s': %v", currentUnixTimestamp, err)
+	}
+
+	now := time.Unix(int64(currentUnixTimestampParsed), 0)
+	// Round current time down to nearest blocksize (2h) and then decrease blocksize (2h)
+	startInclusive := now.Truncate(blockSize).Add(-blockSize)
+	// Round current time down to nearest blocksize (2h) and then add blocksize (2h)
+	endExclusive := now.Truncate(blockSize).Add(blockSize * 2)
```

Comment:
would it be better to take a start/end time (or start time + duration)?

Reply:
Does it not need to be in increments of blocksize?